cores/esp8266/Stream.cpp

```diff
@@ -288,6 +289,59 @@ String Stream::readStringUntil(const char* terminator, uint32_t untilTotalNumber
     return ret;
 }
 
+String Stream::readStreamString(const ssize_t maxLen ,const oneShotMs::timeType timeoutMs) {
+    String ret;
+    S2Stream stream(ret);
+    sendGeneric(&stream, maxLen, -1, timeoutMs);
+    return ret;
+}
+
+String Stream::readStreamStringUntil(const int readUntilChar, const oneShotMs::timeType timeoutMs) {
+    String ret;
+    S2Stream stream(ret);
+    sendGeneric(&stream, -1, readUntilChar, timeoutMs);
+    return ret;
+}
+
+String Stream::readStreamStringUntil (const char* terminatorString, uint32_t untilTotalNumberOfOccurrences, const oneShotMs::timeType timeoutMs) {
```

Comment:
For full genericity with any kind of  stream, we should rather extend [`::sendGeneric`](https://github.com/esp8266/Arduino/blob/0301465a3c53fb42986814f81532b512d47d37fd/cores/esp8266/StreamSend.cpp#L65)  to accept a `char*` instead of a `char`.

Reply:
We Should Do that, but I am having a hard time making this  modification.
This Function will be much simpler if we can do that.
But The basic Idea here is to match the API of `readStringUntil` for single char in the first function and Terminator String in this function.

Reply:
> For full genericity with any kind of stream, we should rather extend [::sendGeneric](https://github.com/esp8266/Arduino/blob/0301465a3c53fb42986814f81532b512d47d37fd/cores/esp8266/StreamSend.cpp#L65) to accept a char* instead of a char.

Having tried implementing that...
- we'd have to change API from char to cstr\*?
- 'basic' streams have to have some kind of logic jump to stall writes until delimiter is actually read, since the current one simply pushes char-by-char
- 'peekBuffer' streams don't have the issue above, but still have to have some kind of extra state management to track delimiter

Still, original Stream methods could be improved? idk if users care about internals here, only that String is the end result.
i.e. https://github.com/esp8266/Arduino/compare/master...mcspr:esp8266-Arduino:strings-webserver/pr9005 as a small experiment where we don't care about the type of Stream, and utilize String buffer as a window into delimited data
(which btw would extend to some other internal libs, not only webserver)